### ISSUE TITLE
Training time is long?

### ISSUE THREAD

**Issue by qqsh0214** [dbolya/yolact#8]
Hi, dbolya.

Thanks for your work. I tried to reproduce the performance with ResNet50 pre-trained model and used the command 'python train.py --config=yolact_resnet50_config'. While training, I found that it need about 30 days to finish the training which was too long. Then I set batch_size = 32 because I have 8 GPUs, but it remains the same. The total training time was still about 30 days. 

Did I do anything wrong? Or the training time is actually long? How can I use Multi-GPU to accelerate training? 

Thanks!

**Comment by dbolya** [comment]
See #3 for training with multiple GPUs. Long story short, use `--no_jit` until I fix the problem. Since you have a batch size of 4 per GPU you may also want to turn off batch norm as in #3.

Sadly, I don't have the resources to figure out the proper number of training iterations and batch size for fast convergence on 4-8 GPUs, so you might have to tweak the number of iterations yourself (but just doing what's in #3 should be enough to get the training time down to ~1 day). Let me know if you are able to get convergence faster than that (`train.py` will print out validation mAP every 2 epochs).

**Comment by qqsh0214** [comment]
Thanks! It really works to get convergence faster than before using `--no_jit.` Now I use 4 GPUs to train with batch_size = 32 so that I don't turn off batch norm. I plan to train 200000 iterations and see the training time and performance.

**Comment by dbolya** [comment]
Remember to also adjust the learning rate decay schedule accordingly [on this line](https://github.com/dbolya/yolact/blob/0d9f1dd7ee8ee7d9a92d4d0089915c7fa8d9edd0/data/config.py#L1645). If you're training for 200k iterations, then `(70000, 150000, 175000, 187500)` is probably a good schedule (dividing everything by 4). Also change `'max_iter'` to 200k. Changing these settings in the base config will update all yolact_* configs.

**Comment by qqsh0214** [comment]
Thanks for reminding!

**Comment by txytju** [comment]
> See #3 for training with multiple GPUs. Long story short, use --no_jit until I fix the problem. Since you have a batch size of 4 per GPU you may also want to turn off batch norm as in #3.

If the batch size is 32 then I think that Batch Norm would work fine? Why did you count number of image per GPU when deciding whether to use Batch Norm? Did I misunderstood the usesage of Batch Norm?

**Comment by dbolya** [comment]
I haven't implemented synchronized batch norm, so batch statistics are only computed on each GPU individually and not on the larger batch. So if you use 8 GPUs with a total batch size of 32, then that would mean that each GPU gets a batch size of 4 for batch norm calculations which would be bad.

4 GPUs with a total batch size of 32 works fine because each GPU gets a batch size of 8, which is what we use to train with batch norm on one GPU.

**Comment by remindchobits** [comment]
Hello!
Is the learning rate decay 0.1 as default or some other value? 
Thx!

**Comment by dbolya** [comment]
@remindchobits Yup lr decay (gamma) is 0.1 by default.

**Comment by remindchobits** [comment]
@dbolya 
Thank you for such a quick reply!
